src/dsp_tools/commands/xmlupload/resource_create_client.py

```diff
@@ -452,28 +458,37 @@ def _make_time_value(value: XMLValue) -> dict[str, Any]:
     }
 
 
-def _serialise_uri_prop(prop: XMLProperty, permissions_lookup: dict[str, Permissions]) -> dict[str, Any]:
-    uri_values: list[SerialiseValue] = []
-    for v in prop.values:
-        if v.permissions:
-            if not (per := permissions_lookup.get(v.permissions)):
-                raise PermissionNotExistsError(f"Could not find permissions for value: {v.permissions}")
-            permission = str(per)
-        else:
-            permission = None
-        uri = cast(str, v.value)
-        uri_values.append(
-            SerialiseURI(
-                value=uri,
-                permissions=permission,
-                comment=v.comment,
-            )
-        )
+def _transform_into_serialise_property_with_string_value(
+    prop: XMLProperty,
+    permissions_lookup: dict[str, Permissions],
+    func: Callable[[str, str | None, str | None], SerialiseValue],
```

Comment:
This is not a function that you call, but a constructor of a class. I think "serialiser" is a good name.